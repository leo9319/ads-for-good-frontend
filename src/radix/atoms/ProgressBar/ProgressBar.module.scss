.progressBarContainer {
  width: 100%;
  max-height: 8px !important;

  div {
    background-color: var(--progressBarColor) !important;
  }
}
