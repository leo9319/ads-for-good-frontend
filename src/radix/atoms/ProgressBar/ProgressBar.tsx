import React from 'react';
import styles from './ProgressBar.module.scss';
import {
  Progress as RadixProgress,
  ProgressProps as Props,
} from '@radix-ui/themes';
import { classNames, getModuleClasses } from '@utils/common/classNames';
import { getStylesAsCssProperties } from '@utils/common/styles';

/**
 * @interface
 * The Progress bar props type of {@link ProgressBar}
 *
 * @property {Props} props - The props of the Progress bar component
 */
export type ProgressProps = Props;

/**
 *
 * Progress bar component to display the Progress bar.
 *
 * @group Atoms
 * @category Radix Component
 *
 * @param {ProgressProps} props - The Progress bar props
 * @returns {ReactElement} - Progress bar Component
 *
 */

export const ProgressBar = ({ className, color, ...props }: ProgressProps) => {
  const classes = getModuleClasses(className?.trim(), styles);
  return (
    <RadixProgress
      className={classNames(styles.progressBarContainer, classes)}
      style={getStylesAsCssProperties({ '--progressBarColor': color })}
      {...props}
    ></RadixProgress>
  );
};

export default ProgressBar;
