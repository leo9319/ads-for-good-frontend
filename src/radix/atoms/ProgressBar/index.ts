export { default } from './ProgressBar';
export { ProgressBar } from './ProgressBar';
export type { ProgressProps } from './ProgressBar';
