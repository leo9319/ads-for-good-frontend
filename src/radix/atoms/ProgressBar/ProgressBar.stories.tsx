import React from 'react';
import type { Meta, StoryObj } from '@storybook/react';
import ProgressBar from './ProgressBar';
import { Theme } from '@radix-ui/themes';

const meta = {
  title: 'Library/Atoms/ProgressBar',
  component: ProgressBar,
  parameters: {
    layout: 'center',
  },
  tags: ['autodocs'],
  decorators: [
    Story => (
      <Theme>
        <Story />
      </Theme>
    ),
  ],
  argTypes: {
    size: {
      control: 'text',
    },
    value: {
      control: 'number',
    },
    color: {
      control: 'color',
    },
    variant: {
      control: 'select',
      options: ['classic', 'surface', 'soft'],
    },
  },
} satisfies Meta<typeof ProgressBar>;

export default meta;
type Story = StoryObj<typeof meta>;

export const progressBar: Story = {
  args: {
    size: '3',
    color: 'orange',
    value: 33,
    variant: 'surface',
  },
};
