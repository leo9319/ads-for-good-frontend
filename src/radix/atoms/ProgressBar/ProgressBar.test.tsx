import React from 'react';
import ProgressBar from './ProgressBar';
import { render } from '@testing-library/react';

describe('ProgressBar', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<ProgressBar />);
    expect(baseElement).toMatchSnapshot();
  });
});
