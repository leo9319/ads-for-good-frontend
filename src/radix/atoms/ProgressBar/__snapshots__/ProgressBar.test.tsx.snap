// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar should render successfully 1`] = `
<body>
  <div>
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      class="rt-ProgressRoot rt-r-size-2 rt-variant-surface progressBarContainer"
      data-max="100"
      data-state="indeterminate"
      role="progressbar"
    >
      <div
        class="rt-ProgressIndicator"
        data-max="100"
        data-state="indeterminate"
      />
    </div>
  </div>
</body>
`;
